arangod/ClusterEngine/ClusterIndex.cpp

```diff
@@ -123,6 +130,66 @@ void ClusterIndex::toVelocyPack(
     builder.add(StaticStrings::IndexEstimates, VPackValue(false));
   }
 
+  if (Index::hasFlag(flags, Index::Serialize::Figures)) {
+    double progress = 0;
+    double success = 0;
+    auto const shards = _collection.shardIds();
+    auto const body = VPackBuffer<uint8_t>();
+    auto* pool =
+        _collection.vocbase().server().getFeature<NetworkFeature>().pool();
+    std::vector<Future<network::Response>> futures;
+    futures.reserve(shards->size());
+    std::string const prefix = "/_api/index/";
+    for (auto const& shard : *shards) {
+      network::RequestOptions reqOpts;
+      reqOpts.param("withHidden", "true");
+      // best effort. only displaying progress
+      reqOpts.timeout = network::Timeout(10.0); // jsteeman, ideas?
```

Comment:
RequestOptions can be created outside of the loop, as they are the same for every shard that is contacted.
```suggestion
    std::string const prefix = "/_api/index/";
    network::RequestOptions reqOpts;
    reqOpts.param("withHidden", "true");
    // best effort. only displaying progress
    reqOpts.timeout = network::Timeout(10.0);    
    for (auto const& shard : *shards) {
```